pkg/plugins/apikey-injection/auth/sigv4_auth_generator.go

```
@@ -0,0 +1,129 @@
+		SessionToken:    credentialsData[awsSessionTokenField],
+		Source:          "ExternalModelSecret",
+	}
+
+	bodyHash := sha256Hex([]byte(body))
+
+	path := credentialsData[enrichedPathField]
+	if path == "" {
+		path = defaultPath
+	}
+
+	req, err := http.NewRequest(http.MethodPost, "https://"+endpoint+path, strings.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)